src/providers/delimitedtext/qgsdelimitedtextprovider.h

```diff
@@ -227,6 +227,7 @@ class QgsDelimitedTextProvider : public QgsVectorDataProvider
     QString mWktFieldName;
     QString mXFieldName;
     QString mYFieldName;
+    bool mDetectTypes;
```

Comment:
Initialise here

Reply:
@nyalldawson Thanks for reviewing .. I'll amend with your suggestion.  While I was looking at it I also noticed that the CSVT interpreter has a very restricted list of types compared to GDAL now.  I've more or less coded this up now.  I was thinking of putting this in a separate pull request, but it could add to this.  Which do you think is preferable?

Reply:
I'd go a separate pr. If it's possible to seperate the work, then it makes reviewing easier

Reply:
Thanks - will do.  Will just sort out autopep for python indentation fixes for this one.

Reply:
No problem! I've tagged with "required changes" for now until this is resolved.